fendermint/vm/interpreter/src/fvm/exec.rs

```diff
@@ -152,21 +150,28 @@ where
     async fn deliver(
         &self,
         mut state: Self::State,
-        mut msg: Self::Message,
+        msg: Self::Message,
     ) -> anyhow::Result<(Self::State, Self::DeliverOutput)> {
         let (apply_ret, emitters, latency) = if msg.from == system::SYSTEM_ACTOR_ADDR {
             let (execution_result, latency) = measure_time(|| state.execute_implicit(msg.clone()));
             let (apply_ret, emitters) = execution_result?;
 
             (apply_ret, emitters, latency)
         } else {
-            // TODO: maybe compare the gas limits is better?
-            msg.gas_limit = msg.gas_limit.min(self.gas.available_block_gas());
+            if msg.gas_limit > state.gas_market().available_block_gas() {
+                bail!("gas limit exceed available block gas limit")
+            }
```

Comment:
Failing here will cause a panic. Can this even happen? Aren't we checking in `process_proposal` that the sum of gas limits does not exceed the maximum block gas limit?

Reply:
maybe we can put a warning log instead as a sanity check.